src/test/integration/PyRegionTest.cpp

```diff
@@ -282,11 +280,53 @@ void testPynodeLinking()
 void testSecondTimeLeak()
 {
   Network n;
-  Network::registerPyRegion("nupic.regions.TestNode", "TestNode");
   n.addRegion("r1", "py.TestNode", "");
   n.addRegion("r2", "py.TestNode", "");
 }
 
+void testFailOnRegisterDuplicateRegion()
+{
+  bool caughtException = false;
+  Network::registerPyRegion("nupic.regions.TestDuplicateNodes", "TestDuplicateNodes");
+  try
+  {
+    Network::registerPyRegion("nupic.regions.TestDuplicateNodes", "TestDuplicateNodes");
+  } catch (std::exception& e) {
+    NTA_DEBUG << "Caught exception as expected: '" << e.what() << "'";
+    caughtException = true;
+  }
+  if (caughtException)
+  {
+    NTA_DEBUG << "testFailOnRegisterDuplicateRegion passed";
+  } else {
+    NTA_THROW << "testFailOnRegisterDuplicateRegion did not throw an exception as expected";
+  }
+}
+
+void testUnregisterRegion()
+{
+  Network n;
+  n.addRegion("test", "py.TestNode", "");
+
+  Network::unregisterPyRegion("TestNode");
+
+  bool caughtException = false;
+  try
+  {
+    n.addRegion("test", "py.TestNode", "");
+  } catch (std::exception& e) {
+    NTA_DEBUG << "Caught exception as expected: '" << e.what() << "'";
+    caughtException = true;
```

Comment:
While it doesn't look like this test has been converted yet, we are moving towards using gtest for C++ which has a nice macro for this:
https://code.google.com/p/googletest/wiki/AdvancedGuide#Exception_Assertions

Example test that has been converted:
https://github.com/numenta/nupic.core/blob/master/src/test/unit/algorithms/FastCLAClassifierTest.cpp